File: backend/src/main/java/tn/esprit/forum/entities/Entretien.java
package tn.esprit.forum.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.LocalDate;
import java.time.LocalTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Entretien {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Setter(AccessLevel.NONE)
    Long idEntretien;
    LocalTime time;
    LocalDate date;
    String room;
}

File: frontend/src/app/core/models/Entretien.ts
export class Entretien {
    idEntretien: number;
    time: string;
    date: string;
    room: string;
  
    constructor(
      idEntretien: number,
      time: string,
      date: string,
      room: string
    ) {
      this.idEntretien = idEntretien;
      this.time = time;
      this.date = date;
      this.room = room;
    }
  }
  